Assets/MixedRealityToolkit.SDK/Features/UX/Interactable/Scripts/Interactable.cs

```diff
@@ -127,6 +127,38 @@ protected static IMixedRealityInputSystem InputSystem
         /// </summary>
         public bool FocusEnabled { get { return !IsGlobal; } set { IsGlobal = !value; } }
 
+        /// <summary>
+        /// Event receivers can be used to listen for different
+        /// events at runtime. This method allows receivers to be dynamically added at runtime.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <returns>The new event receiver</returns>
+        public T AddReceiver<T>() where T : ReceiverBase, new()
+        {
+            var interactableEvent = new InteractableEvent();
+            var result = new T();
+            result.Event = interactableEvent.Event;
+            interactableEvent.Receiver = result;
+            Events.Add(interactableEvent);
+            return result;
+        }
+
+        /// <summary>
+        /// Returns the first receiver of type T on the interactable,
+        /// or null if nothing is found.
+        /// </summary>
+        public T GetReceiver<T>() where T : ReceiverBase
```

Comment:
>GetReceiver [](start = 17, length = 11)

It's also odd to me that it returns the first found of a certain type (meaning if I add two of the same type, it will never return the other one) - this particular type of API surface (first one wins) is of an odd design (usually you want to provide deterministic way of getting a specific instance).

i.e. if I call AddReciever twice and then lose the second one, there's no way for me to get the second one using these typical APIs (I guess I could iterate over the events).

Maybe there should be an overload that returns all that match (keeping the singular case for convenience/typical usage)? Or should there never be multiples there?

Reply:
I added overload for returning all matches.

Reply:
But actually it seems odd that you'd have multiple of a type. But, it is possible so I added an overload.